assets/js/components/ImgDropAndCrop/index.js

```diff
@@ -0,0 +1,122 @@
+                return false
+            }
+            return true
+        }
+    }
+
+    handleOnDrop = (files, rejectedFiles) => {
+        if (rejectedFiles && rejectedFiles.length > 0) {
+            this.verifyFile(rejectedFiles)
+        }
+
+        if(files && files.length > 0) {
+            const isVerified = this.verifyFile(files)
+            if (isVerified) {
+                const currentFile = files[0]
+                const myFileItemReader = new FileReader()
+                myFileItemReader.addEventListener('load', () => {
+                    //console.log(myFileItemReader.result)
+                    this.setState({
+                        imgSrc: myFileItemReader.result,
+                    })
+                }, false)
+
+                myFileItemReader.readAsDataURL(currentFile)
+            }
+        }
+    }
+
+    handleImageLoaded = (image) => {
+        //console.log(Image)
+    }
+
+    handleOnCropChange = (crop) => {
+        this.setState({crop:crop})
+    }
+
+    handleOnCropComplete = (crop, pixelCrop) => {
+        console.log(crop, pixelCrop)
+        const canvasRef = this.imagePreviewCanvasRef.current
+        const {imgSrc} = this.state
+        image64toCanvasRef(canvasRef, imgSrc, pixelCrop)
+    }
+
+    handleOnCropUpload = (event) => {
+        event.preventDefault()
+        const {imgSrc} = this.state
+        const canvasRef = this.imagePreviewCanvasRef.current
+        const fileExtension = extractImageFileExtensionFromBase64(imgSrc)
+        const imageData64 = canvasRef.toDataURL('/image' + fileExtension)
+        const myFilename = 'User1' + fileExtension
+        console.log(myFilename)
+        //const myNewCroppedFile = base64StringtoFile(imageData64, myFilename)
+        this.props.getCroppedImage(imageData64);
+    }
+
+
+    render() {
+        const {imgSrc} = this.state
+        return(
+            <div>
+                {imgSrc !== null ?
+                    <div>    
+                        <ReactCrop 
+                            src={imgSrc}
+                            crop={this.state.crop}
+                            onImageLoaded={this.handleImageLoaded}
+                            onComplete={this.handleOnCropComplete} 
+                            onChange={this.handleOnCropChange}/>
+                        <br />
+                        <p>Preview Canvas Crop</p>
+                        <canvas ref={this.imagePreviewCanvasRef} style={{maxWidth: '135px', display: 'none'}}></canvas>
+                    </div> 
+                    : 
+                    <Dropzone
+                        onDrop={this.handleOnDrop}
+                        multiple={false}
+                        maxSize={imageMaxSize}
+                        accept={acceptedFileTypes} 
+                    >Drop image here / Click to upload</Dropzone>
+                }
+                <button onClick={this.handleOnCropUpload}>Save Cropped Image</button>
```

Comment:
I think this button should only be visible when an image is being cropped.
The text should be translatable.